runs/restore.sh

```diff
@@ -1,20 +1,31 @@
 #!/bin/bash
-echo "****************************************"
-echo "Step 1: moving file to home directory..."
-sudo cp /var/www/html/openWB/web/tools/upload/backup.tar.gz /home/pi/backup.tar.gz
-cd /home/pi/
-echo "****************************************"
-echo "Step 2: extracting archive..."
-sudo tar -vxf backup.tar.gz
-echo "****************************************"
-echo "Step 3: replacing old files..."
-sudo cp -v -R /home/pi/var/www/html/openWB/* /var/www/html/openWB/
-sudo chmod 777 /var/www/html/openWB/openwb.conf
-echo "****************************************"
-echo "Step 4: cleanup after restore..."
-sudo rm /var/www/html/openWB/web/tools/upload/backup.tar.gz
-sudo rm /home/pi/backup.tar.gz
-sudo rm -R /home/pi/var
-echo "****************************************"
-echo "End: Restore finished."
-echo "****************************************"
+OPENWBBASEDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+SOURCEFILE="$OPENWBBASEDIR/web/tools/upload/backup.tar.gz"
+WORKINGDIR="/home/pi/openwb_restore"
+LOGFILE="$OPENWBBASEDIR/web/tools/upload/restore.log"
+
+{
+	echo "$(date +"%Y-%m-%d %H:%M:%S") Restore of backup started..."
+	echo "****************************************"
+	echo "Step 1: creating working directory \"$WORKINGDIR\""
+	mkdir -p "$WORKINGDIR"
+	echo "****************************************"
+	echo "Step 2: extracting archive to working dir \"$WORKINGDIR\"..."
+	sudo tar -vxf "$SOURCEFILE" -C "$WORKINGDIR"
+	echo "****************************************"
+	echo "Step 3: replacing old files..."
+	cp -v -R -p "${WORKINGDIR}${OPENWBBASEDIR}/." "${OPENWBBASEDIR}R/"
+	echo "****************************************"
+	echo "Step 4: restoring mosquitto db..."
+	sudo systemctl stop mosquitto.service
+	sleep 2
+	sudo cp -v -p "$WORKINGDIR/var/lib/mosquitto/mosquitto.db" "/var/lib/mosquitto/mosquitto.db"
+	sudo systemctl start mosquitto.service
+	echo "****************************************"
+	echo "Step 5: cleanup after restore..."
+	sudo rm "$SOURCEFILE"
+	sudo rm -R "$WORKINGDIR"
```

Comment:
Das ginge auch in einem Befehl:
```bash
sudo rm -R "$SOURCEFILE" "$WORKINGDIR"
```
spart einen Prozessstart ein und ist etwas kompakter.

Reply:
oder gleich zu 
'''
sudo rm -R /
'''
:-) ist noch kürzer

Ich wurde immer mit einem Konstanten Bestandteil arbeiten, nicht mit Variablen (die auch "/" enthalten können)

Also 
sudo rm -r "/home/pi/$Workdir" "/var/www/html/openWB/web/files/upload/$SOURCEFILE"
Da brennt nix unwiederbringlich an wenn die Variablen mal murks enthalten.

Reply:
Und was ist in deinem Beispiel, wenn `$Workdir=../../` ist? Dann löschst du doch wieder alles.

Sinnvoller wäre das ganze nicht als `sudo` zu machen. Aber aus historischen Gründen wird das leider gebraucht um ownership zu erhalten...

Die gewählte Variante mit den Variablen ist schon sinnvoll, denn sie garantiert, dass alle Befehle, die mit dem entsprechenden Pfad arbeiten auch definitiv den gleichen Pfade haben. Das Prefix hard zu codieren ist nicht sinnvoll, weil du dich der Möglichkeit beraubst ein anderes Prefix zu verwenden.

Reply:
> leider gebraucht um ownership zu erhalten...
Bei rm?

Reply:
> Bei rm?

Nein,  bei dem `tar`. Aber dadurch haben die Dateien teilweise nicht `pi` als owner und deswegen braucht das `rm` dann auch `sudo`.

Reply:
Ich werde die beiden Aufrufe so drin lassen. DAmit ist zumindest direkt sichtbar, dass nur beim zweiten Befehl rekursiv gelöscht wird. Einen riesigen Nachteil dürfte das nicht haben.